### ISSUE TITLE
ISW - Category and Illuminate Pack for inputs

### ISSUE THREAD

**Issue by grotlue** [Graylog2/graylog2-server#20535]
- API for categories (for a specific category)

**Comment by grotlue** [comment]
We will need an API for categories and subcategories (for a specific category). This will also be needed for the Data Source wizard.

**Comment by AntonEbel** [comment]
This are the 
- [cats subcats](https://graylogdocumentation.atlassian.net/wiki/spaces/PROD/pages/3494019099/Attack+Surface+Areas+Visibility+Scores#Attack-Surface-Groups) 
- plus "generic" subcat

**Comment by grotlue** [comment]
For this cycle adding categories and subcategories in the UI is out of scope.

**Comment by patrickmann** [comment]
Illuminate packs will include the data required by the input setup wizard. Specifically, the following fields: 
- `category`: the category name, as it will be shown in the UI
- `subcategory`: the subcategory name, as it will be shown in the UI
- `subcategory_description`: detailed description
- `input_types`: list of all associated input types

E.g. the pack for Cisco Checkpoint Firewall would look like this:
```
[[packs]]
id = "3d009884-98a8-4081-8a73-f97d1bffe7fc"
title = "Illuminate %%VERSION%%:Checkpoint Firewall"
description = """
Checkpoint Next Gen Firewall (NGFW) features include Firewall, DLP, threat protection, virus protection, VPN management. 
This Illuminate pack provides normalization, enrichment, and categorization of Checkpoint NGFW events.
"""
requirements = [ "core" ]
[...]
editions = ["cloud-security", "standard-security", "development", "cloud-operations", "standard-operations"]
category = "Firewall"
subcategory = "Checkpoint Firewall"
subcategory_description = """
The Cisco Firepower range are cybersecurity products designed ....
"""
input_types = ["syslog"]
```

**Comment by grotlue** [comment]
Will it also include information regarding associated licenses?

**Comment by patrickmann** [comment]
@grotlue Yes - Illuminate pack includes `editions`, which tells us if it is available in open/enterprise/security and cloud/on-prem. E.g.:
https://github.com/Graylog2/graylog-project-illuminate/blob/develop/_meta/packs/bind_dns.toml

**Comment by tellistone** [comment]
We should add an “other” category. If user selects this, we present them with all (valid) illuminate packs.

**Comment by patrickmann** [comment]
Obsolete, since Illuminate is providing the required APIs
